src/lib/operators/table_scan.cpp

```diff
@@ -172,11 +174,37 @@ std::shared_ptr<const Table> TableScan::_on_execute() {
   return output_table;
 }
 
+std::shared_ptr<AbstractExpression> TableScan::_resolve_uncorrelated_subqueries(
+    const std::shared_ptr<AbstractExpression>& predicate) {
+  // If the predicate has an uncorrelated subquery on the top level, we resolve that subquery first. That way, we can
```

Comment:
If we go for generality, let's go all in: Use visit_expression to crawl the entire expression for uncorrelated subselects. `visit_expression` is written so that you can replace individual sub_expressions on the fly. 
You can also avoid the deep copy if you check for use_count(), I guess.

Reply:
Even if you decide against this, "on the top level" is not what this covers. You're scanning the arguments of the top-level expression, after all.

Reply:
Visiting the subexpressions does not make sense - if we have any subexpressions, we need the ExpressionEvaluator anyway.

Reply:
Right! Can you doc that, too?